### PR TITLE
C++11 feature: std::async

### PR DESCRIPTION
**Pull request by defmys**
Issue #46

### CODE REVIEW

File: CPP11.md [hunk 1]
```diff
@@ -738,6 +739,27 @@ C++11 introduces a memory model for C++, which means library support for threadi
 
 See the sections on: [std::thread](#stdthread)
 
+### std::async
+`std::async` runs function asynchronously or perform lazy evaluation, then returns a `std::future` which holds the result of that function call.
```
Comment (AnthonyCalandra) [inline CPP11.md]: `runs the given function either asynchronously or lazily-evaluated`*
Just some minor tweaks here.

File: CPP11.md [hunk 2]
```diff
@@ -738,6 +739,27 @@ C++11 introduces a memory model for C++, which means library support for threadi
...
+
+The first parameter is the policy which can be:
+1. `std::launch::async | std::launch::deferred` It is up to the implementation whether to perform asynchronous execution or lazy evaluation.
+1. `std::launch::async` Run the callable object on a new thread.
+1. `std::launch::deferred` Perform lazy evaluation on the current thread.
+
+```
+int foo() {
+    /*do something here, then return the result*/
```
Comment (AnthonyCalandra) [inline CPP11.md]: `/* Do something here, then return the result. */`

File: CPP11.md [hunk 3]
```diff
@@ -738,6 +739,27 @@ C++11 introduces a memory model for C++, which means library support for threadi
...
+
```
Comment (AnthonyCalandra) [inline CPP11.md]: Remove the extra newline.

File: CPP11.md [hunk 4]
```diff
@@ -738,6 +739,27 @@ C++11 introduces a memory model for C++, which means library support for threadi
...
+1. `std::launch::deferred` Perform lazy evaluation on the current thread.
+
+```
+int foo() {
+    /*do something here, then return the result*/
+    
+    return 1000;
+}
+
+
```
Comment (AnthonyCalandra) [inline CPP11.md]: Remove the extra newline.

File: CPP11.md [hunk 5]
```diff
@@ -738,6 +739,27 @@ C++11 introduces a memory model for C++, which means library support for threadi
...
+auto handle = std::async(std::launch::async, foo);  // create an async task
+auto result = handle.get();  // wait for the result
+
```
Comment (AnthonyCalandra) [inline CPP11.md]: Remove the extra newline.

**Comment by defmys** [comment]
Thank you for your review. I have committed those changes.

**Pull request merged**